test/test_task_manager.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+
+from openwpm.errors import CommandExecutionError
+from openwpm.task_manager import TaskManager
+
+from .openwpmtest import OpenWPMTest
+
+
+class TestTaskManager(OpenWPMTest):
+    """Test TaskManager functionality."""
+
+    def get_config(self, data_dir=""):
+        return self.get_test_config(data_dir)
+
+    def test_failure_limit_value(self):
+        manager_params, _ = self.get_config()
+        # The default value for failure_limit is 2 * num_browsers + 10
+        assert manager_params.failure_limit == 12
+        manager_params.failure_limit = 2
+        # Test that the chosen value is not overwritten by the default
+        assert manager_params.failure_limit == 2
+
+    def test_failure_limit_exceeded(self):
+        manager_params, browser_params = self.get_config()
+        manager_params.failure_limit = 0
+        manager = TaskManager(manager_params, browser_params)
+        with pytest.raises(CommandExecutionError):
+            manager.get("example.com")  # Selenium requires scheme prefix
+            manager.get("example.com")  # Requires two commands to shut down
```

Comment:
(Not saying it should work this way, I'm just unfamiliar with the surrounding code)

Reply:
I think it is because `_check_failure_status()`, which raises the `CommandExecutionError` is called [here](https://github.com/mozilla/OpenWPM/blob/078395c4ad148a60e20a6dbac549a2287fde2cc3/openwpm/task_manager.py#L375), before executing each command sequence, and not after. However, if the command sequence is blocking, `_check_failure_status()` is also called [here](https://github.com/mozilla/OpenWPM/blob/078395c4ad148a60e20a6dbac549a2287fde2cc3/openwpm/task_manager.py#L665) and in this case we don't need to run a second command sequence for the error to be raised.

Reply:
Interesting. Would you mind adding that as a second point to the docs?

Reply:
Not at all. I updated the docs. Please let me know if the wording is clear enough.

Reply:
Thanks! The wording looks good to me.